pkg/k8s/resource/resource.go

```diff
@@ -148,35 +181,33 @@ func (r *resource[T]) Start(startCtx hive.HookContext) error {
 
 	store, informer := cache.NewInformer(r.lw, objType, 0, handlerFuncs)
 
-	r.wg.Add(2)
+	r.wg.Add(1)
 	go func() {
 		defer r.wg.Done()
 		informer.Run(r.ctx.Done())
 	}()
-	go func() {
-		defer r.wg.Done()
 
-		// Wait for cache to be synced before resolving the store
-		if !cache.WaitForCacheSync(r.ctx.Done(), informer.HasSynced) {
-			// The context is cancelled when stopping and all dependees of Resource[T] are
-			// stopped before it, but to be safe, resolve the store with nil to catch
-			// misbehaving dependees.
-			r.storeResolver.Reject(r.ctx.Err())
-			return
-		}
+	// Wait for cache to be synced before resolving the store
+	if !cache.WaitForCacheSync(r.ctx.Done(), informer.HasSynced) {
+		// The context is cancelled when stopping and all dependees of Resource[T] are
+		// stopped before it, but to be safe, resolve the store with nil to catch
+		// misbehaving dependees.
+		r.storeResolver.Reject(r.ctx.Err())
+	} else {
 		r.storeResolver.Resolve(&typedStore[T]{store})
-	}()
-
-	return nil
+	}
 }
 
 func (r *resource[T]) Stop(stopCtx hive.HookContext) error {
 	r.cancel()
+	r.markNeeded()
```

Comment:
Not sure if I like this. Now we have to close the context and signal the sync.Cond to get goroutines to stop, instead of a hard rule that canceling the context will stop all associated routines.

Reply:
I would've preferred something simpler as well. Anything come to mind?

Reply:
Reimplemented using a buffered channel, so now it can select over ctx.Done and the 'needed' channel. Is this better?

Reply:
This seems like a better way to do it.